### PR TITLE
[BACKLOG-36855] : WCAG Opened Perspective

### PR DESCRIPTION
**Pull request by addagudi**

### CODE REVIEW

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 26s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **333**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-837/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 7m 2s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **333**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-837/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: widgets/src/main/java/org/pentaho/gwt/widgets/client/tabs/PentahoTab.java [hunk 1]
```diff
@@ -41,6 +41,7 @@ public class PentahoTab extends SimplePanel {
   private Widget content;
   protected Label label = new Label();
   private boolean solutionBrowserShowing;
+  private boolean closeable;
```
Comment (dcleao) [inline widgets/src/main/java/org/pentaho/gwt/widgets/client/tabs/PentahoTab.java:44]: Please, define a corresponding `public boolean isCloseable()` method (and use that instead elsewhere in this file).

File: widgets/src/main/java/org/pentaho/gwt/widgets/client/tabs/PentahoTab.java [hunk 2]
```diff
@@ -121,6 +123,10 @@ public void onBrowserEvent( Event event ) {
         moveFocusToPreviousTab();
       } else if ( KeyCodes.KEY_RIGHT == event.getKeyCode() ) {
         moveFocusToNextTab();
+      } else if ( this.closeable && KeyCodes.KEY_DELETE == (char) event.getKeyCode() ) {
```
Comment (dcleao) [inline widgets/src/main/java/org/pentaho/gwt/widgets/client/tabs/PentahoTab.java:126]: Please, use `isCloseable()` instead.

File: widgets/src/main/java/org/pentaho/gwt/widgets/client/tabs/PentahoTab.java [hunk 3]
```diff
@@ -41,6 +41,7 @@ public class PentahoTab extends SimplePanel {
   private Widget content;
   protected Label label = new Label();
   private boolean solutionBrowserShowing;
+  private boolean closeable;
 
   public PentahoTab( String text, String tooltip, PentahoTabPanel tabPanel, Widget content, boolean closeable ) {
     this.content = content;
```
Comment (dcleao) [inline widgets/src/main/java/org/pentaho/gwt/widgets/client/tabs/PentahoTab.java:47]: I believe you should remove the recently added `focus-visible` class (line 50) (by @hv-vamsi in https://github.com/pentaho/pentaho-commons-gwt-modules/pull/824).

https://github.com/pentaho/pentaho-commons-gwt-modules/blob/0d000bb0f2493565ec17e04a2aaf842c9a872965/widgets/src/main/java/org/pentaho/gwt/widgets/client/tabs/PentahoTab.java#L49

As well as a similar reference in https://github.com/hv-vamsi/pentaho-commons-gwt-modules/blob/00561f6ae56b4ddc4f77728a6dda1690d0e8d665/widgets/src/main/java/org/pentaho/gwt/widgets/client/buttons/ThemeableImageButton.java#L105.

File: widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css [hunk 4]
```diff
@@ -1189,6 +1189,11 @@ button#schedule-dialog-select-button {
   border-right: solid 1px transparent;
 }
 
+.pentaho-tabWidget:focus-visible {
+  outline: 3px solid #0075ff;
+  outline-offset: -2px;
+}
+
```
Comment (dcleao) [inline widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css:1196]: This is the default focus ring style. Should not be needed. See https://github.com/hv-vamsi/pentaho-commons-gwt-modules/blob/a1828a129741e00f4f67a504d1076c04baadd315/widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css#L43.

File: widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css [hunk 5]
```diff
@@ -1675,6 +1680,10 @@ table.toolbar {
   margin-left: 5px !important;
 }
 
+.toolbar-button-focus-panel:focus-visible {
+  outline: 3px solid #0075ff;
+}
+
```
Comment (dcleao) [inline widgets/src/main/resources/org/pentaho/gwt/widgets/themes/public/themes/ruby/globalRuby.css:1686]: Again, AFAICT, should not be needed.

**Review by dcleao — CHANGES_REQUESTED** [review]

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 25s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **333**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-837/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: widgets/src/main/java/org/pentaho/gwt/widgets/client/tabs/PentahoTab.java (hunk 1 above)
Comment (addagudi) [reply widgets/src/main/java/org/pentaho/gwt/widgets/client/tabs/PentahoTab.java:44]: Done

File: widgets/src/main/java/org/pentaho/gwt/widgets/client/tabs/PentahoTab.java (hunk 3 above)
Comment (addagudi) [reply widgets/src/main/java/org/pentaho/gwt/widgets/client/tabs/PentahoTab.java:47]: Done

File: widgets/src/main/java/org/pentaho/gwt/widgets/client/tabs/PentahoTab.java (hunk 2 above)
Comment (addagudi) [reply widgets/src/main/java/org/pentaho/gwt/widgets/client/tabs/PentahoTab.java:126]: Done

File: widgets/src/main/java/org/pentaho/gwt/widgets/client/toolbar/ToolbarButton.java [hunk 6]
```diff
@@ -12,19 +12,13 @@
...
 
-import com.google.gwt.event.dom.client.MouseDownEvent;
-import com.google.gwt.event.dom.client.MouseDownHandler;
-import com.google.gwt.event.dom.client.MouseOutEvent;
-import com.google.gwt.event.dom.client.MouseOutHandler;
-import com.google.gwt.event.dom.client.MouseOverEvent;
-import com.google.gwt.event.dom.client.MouseOverHandler;
-import com.google.gwt.event.dom.client.MouseUpEvent;
-import com.google.gwt.event.dom.client.MouseUpHandler;
+import com.google.gwt.event.dom.client.*;
```
[the remark this answers is not in the document]
Comment (addagudi) [reply widgets/src/main/java/org/pentaho/gwt/widgets/client/toolbar/ToolbarButton.java:20]: Done

File: widgets/src/main/java/org/pentaho/gwt/widgets/client/toolbar/ToolbarButton.java [hunk 7]
```diff
@@ -211,6 +206,31 @@ public void onMouseUp( Widget arg0, int arg1, int arg2 ) {
...
+      }
+    });
+  }
+
+  public HandlerRegistration addKeyDownHandler ( KeyDownHandler handler ) {
+    return this.button.addDomHandler( handler, KeyDownEvent.getType() );
+  }
+
+  public HandlerRegistration addKeyUpHandler ( KeyUpHandler handler ) {
+    return this.button.addDomHandler( handler, KeyUpEvent.getType() );
```
[the remark this answers is not in the document]
Comment (addagudi) [reply widgets/src/main/java/org/pentaho/gwt/widgets/client/toolbar/ToolbarButton.java:233]: Done

File: widgets/src/main/java/org/pentaho/gwt/widgets/client/toolbar/ToolbarButton.java [hunk 8]
```diff
@@ -278,6 +298,7 @@ public boolean isVisible() {
   public void setVisible( boolean visible ) {
     this.visible = visible;
     button.setVisible( visible );
+    eventWrapper.getElement().setTabIndex(visible ? 0 : -1);
```
[the remark this answers is not in the document]
Comment (addagudi) [reply widgets/src/main/java/org/pentaho/gwt/widgets/client/toolbar/ToolbarButton.java:301]: Done

File: widgets/src/main/java/org/pentaho/gwt/widgets/client/toolbar/ToolbarToggleButton.java [hunk 9]
```diff
@@ -12,11 +12,13 @@
  * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
  * See the GNU Lesser General Public License for more details.
  *
- * Copyright (c) 2002-2017 Hitachi Vantara..  All rights reserved.
+ * Copyright (c) 2002-2023 Hitachi Vantara..  All rights reserved.
  */
 
 package org.pentaho.gwt.widgets.client.toolbar;
 
+import com.google.gwt.event.dom.client.*;
```
[the remark this answers is not in the document]
Comment (addagudi) [reply widgets/src/main/java/org/pentaho/gwt/widgets/client/toolbar/ToolbarToggleButton.java:20]: Done

File: widgets/src/main/java/org/pentaho/gwt/widgets/client/toolbar/ToolbarToggleButton.java [hunk 10]
```diff
@@ -202,6 +204,38 @@ public void onMouseUp( Widget arg0, int arg1, int arg2 ) {
...
+    });
+
+  }
+
+  public HandlerRegistration addKeyDownHandler (KeyDownHandler handler ) {
+    return this.button.addDomHandler( handler, KeyDownEvent.getType() );
+  }
+
+  public HandlerRegistration addKeyUpHandler ( KeyUpHandler handler ) {
+    return this.button.addDomHandler( handler, KeyUpEvent.getType() );
```
[the remark this answers is not in the document]
Comment (addagudi) [reply widgets/src/main/java/org/pentaho/gwt/widgets/client/toolbar/ToolbarToggleButton.java:238]: Done

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 6m 5s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **333**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-837/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 5m 49s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **333**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-837/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 1h 25m 23s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **333**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-837/7/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — CHANGES_REQUESTED** [review]

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 5m 47s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **333**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-837/8/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 5m 44s
Build command:
```console
mvn clean verify -B -e -Daudit -amd -pl widgets
```
---
:ok_hand: All tests passed!
 
Tests run: **333**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/gwt-widgets/job/PR-837/9/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by dcleao — APPROVED** [review]

**Review by hv-vamsi — APPROVED** [review]

**Pull request merged**